### DISCUSSION TITLE
Hook launching rules in Vike with clientRendering

### DISCUSSION THREAD

**Discussion by igolkotek** [vikejs/vike#1431] [Discussion]
Hi,
This is not a complaint or asking for an idea.
Just depicting the current Vike rules for the following hooks:
- +guard.js
- +data.js
- +onBeforeRender.js

Configuration:
```js
{
  clientRendering: true,
  meta: {
    onBeforeRender: {
      env: { server: true, client: true }
    },
    guard: {
      env: { server: false, client: true }
    },
    data: {
      env: { server: false, client: true }
    }
  }
}
```
At the server side the hooks are run in the following order:
- guard
- data
- onBeforeRender

At the client side the order for the 1st page is as following:
- data

On every next page the client fires the hooks as on the server side:
- guard
- data
- onBeforeRender

Not sure it sounds good for the documentation.

**Comment by brillout** [comment]
Thanks for pushing for that. I completely agree that we should have a graph showing the order. I've created a ticket for it: https://github.com/vikejs/vike/issues/1435. Stay tuned.

**Reply by igolkotek** [reply]
Great, thank you
